photon-client/src/integTest/scala/com/linkedin/photon/ml/data/avro/AvroDataWriterIntegTest.scala

```diff
@@ -0,0 +1,45 @@
+package com.linkedin.photon.ml.data.avro
+
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.testng.Assert._
+import org.testng.annotations.Test
+
+import com.linkedin.photon.ml.io.FeatureShardConfiguration
+import com.linkedin.photon.ml.test.SparkTestUtils
+
+/**
+ * Integeration test for AvroDataWriter
+ */
+class AvroDataWriterIntegTest extends SparkTestUtils {
+
+  import AvroDataWriterIntegTest._
+
+  @Test
+  def testSaveAvro(): Unit = sparkTest("testRead") {
+    val dr = new AvroDataReader()
+    val (df, indexMapLoadersMap) = dr.readMerged(inputPath.toString, featureShardConfigurationsMap, numPartitions)
+
+    assertTrue(df.columns.contains(featureColumn))
+    assertTrue(df.columns.contains(responseColumn))
+    assertEquals(df.count, 34810)
+    assertTrue(indexMapLoadersMap.contains(featureColumn))
+
+    val indexMapLoader = indexMapLoadersMap(featureColumn)
+    AvroDataWriter.saveAvro(sc, df, outputPath, indexMapLoader, responseColumn, featureColumn)
+
+    val fs = FileSystem.get(sc.hadoopConfiguration)
+    val files = fs.listStatus(new Path(outputPath)).filter(_.getPath.getName.startsWith("part"))
+    assertEquals(files.length, numPartitions)
+  }
+}
+
+object AvroDataWriterIntegTest {
+  private val inputDir = getClass.getClassLoader.getResource("GameIntegTest/input").getPath
+  private val inputPath = new Path(inputDir, "train")
+  private val outputPath = "/tmp/AvroDataWriter/"
```

Comment:
Use the `TestTemplateWithTmpDir` mixin to generate a temporary folder instead. It generates the folder in a more OS-portable way, guarantees that access to the folder is thread-safe, and also cleans up the temporary folder after the tests finish.

Reply:
done